### PR TITLE
refactor: menu component token

### PR DESCRIPTION
**Pull request by MadCcc**
- docs: update examples link
- refacor: menu component token

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Migrate Component Token of Menu from 4.x less variables.       |
| 🇨🇳 Chinese |   补全 Menu 的组件 token。        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at ea38a8f</samp>

This pull request introduces new style tokens and refactors the existing style functions for the menu component to enable more customization and consistency. It also updates the documentation to reflect the changes and fixes the `DOCTYPE` declaration in the blog posts.

### <samp>🤖 Generated by Copilot at ea38a8f</samp>

*  Rename `menuHorizontalHeight` to `horizontalLineHeight` and `menuItemPaddingInline` to `itemPaddingInline` for clarity and consistency ([link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-ebae966ac1b7399f66aa822d0ead52d0e27fcc1d5d23dfc05eaf39125e693467L8-R17), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-ebae966ac1b7399f66aa822d0ead52d0e27fcc1d5d23dfc05eaf39125e693467L34-R34), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL253-R349), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL280-R376), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL689-L690), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-803a82d9b4afa684620a38d6c44836f0b24b12205ef42ec80dba78aa28fe9829L24-R24), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-803a82d9b4afa684620a38d6c44836f0b24b12205ef42ec80dba78aa28fe9829L177-R179), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-b7b7775fd3e73c4510f6e127d709629338c94f257ad4cde92ffa63c62f11e25bL9-R14), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-b7b7775fd3e73c4510f6e127d709629338c94f257ad4cde92ffa63c62f11e25bL26-R32))
* Add `groupTitleLineHeight` and `groupTitleFontSize` tokens to customize group title style in `Menu` component ([link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dR34-R43), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dR489-R490), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL447-R545), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL776-R887), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL243-R287), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L246-R285))
* Add `itemHeight`, `collapsedWidth`, `popupBg`, `itemMarginBlock`, `horizontalLineHeight`, `iconSize`, `iconMarginInlineEnd`, `collapsedIconSize`, and dark theme tokens to customize `Menu` component in different modes, themes, and states ([link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dR3), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL232-R333), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL253-R349), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL271-R368), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL280-R376), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dR780-R787), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL700-R833), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL776-R887), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-803a82d9b4afa684620a38d6c44836f0b24b12205ef42ec80dba78aa28fe9829R48-R49), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-803a82d9b4afa684620a38d6c44836f0b24b12205ef42ec80dba78aa28fe9829L158-R160), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-b7b7775fd3e73c4510f6e127d709629338c94f257ad4cde92ffa63c62f11e25bL38-R39), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-b7b7775fd3e73c4510f6e127d709629338c94f257ad4cde92ffa63c62f11e25bL65-R71), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-b7b7775fd3e73c4510f6e127d709629338c94f257ad4cde92ffa63c62f11e25bL163-R165), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-b7b7775fd3e73c4510f6e127d709629338c94f257ad4cde92ffa63c62f11e25bL191-R194), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL243-R287), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L246-R285))
* Remove unused or replaced tokens from `genMenuItemStyle`, `getBaseStyle`, and `genComponentStyleHook` functions ([link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL247), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL382), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL669-R766), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-f84706d0f4fb54c91e83f8dd7b34ce37f6015d56dc057385580b19473c8f1f7dL689-L690))
* Change `DOCTYPE` declaration from uppercase to lowercase in `extract-ssr.en-US.md` and `extract-ssr.zh-CN.md` documents to follow HTML5 standard ([link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-47e91f6d77e2ac334481396f7a6ecdcb0e936712e0aed674a873aeed20d7ca8eL142-R142), [link](https://github.com/ant-design/ant-design/pull/43576/files?diff=unified&w=0#diff-587aeb46cf39b8ffeb466a0e903d4b92aed00f64efafba33e9a63c656fce2df5L142-R142))

### CODE REVIEW

**Review by kiner-tang — APPROVED** [review]

**Review by li-jia-nan — APPROVED** [review]

File: .dumi/theme/slots/Sidebar/index.tsx [hunk 1]
```diff
@@ -148,10 +150,10 @@ const Sidebar: React.FC = () => {
   );
 
   return isMobile ? (
-    <MobileMenu key="Mobile-menu">{menuChild}</MobileMenu>
+    <MobileMenu key='Mobile-menu'>{menuChild}</MobileMenu>
   ) : (
     <Col xxl={4} xl={5} lg={6} md={6} sm={24} xs={24} css={styles.mainMenu}>
-      <section className="main-menu-inner">{menuChild}</section>
+      <section className='main-menu-inner'>{menuChild}</section>
```
Comment (afc163) [inline .dumi/theme/slots/Sidebar/index.tsx:156]: 好像不应该改这个。

File: .dumi/theme/slots/Sidebar/index.tsx [hunk 2]
```diff
@@ -134,12 +134,14 @@ const Sidebar: React.FC = () => {
...
-    <ConfigProvider theme={{ components: { Menu: { itemBg: colorBgContainer } } }}>
+    <ConfigProvider
+      theme={{ components: { Menu: { itemBg: colorBgContainer, darkItemBg: colorBgContainer } } }}
+    >
       <Menu
         items={menuItems}
         inlineIndent={30}
         css={styles.asideContainer}
-        mode="inline"
+        mode='inline'
```
Comment (MadCcc) [inline .dumi/theme/slots/Sidebar/index.tsx:144]:
```suggestion
        mode="inline"
```

File: .dumi/theme/slots/Sidebar/index.tsx [hunk 3]
```diff
@@ -148,10 +150,10 @@ const Sidebar: React.FC = () => {
   );
 
   return isMobile ? (
-    <MobileMenu key="Mobile-menu">{menuChild}</MobileMenu>
+    <MobileMenu key='Mobile-menu'>{menuChild}</MobileMenu>
```
Comment (MadCcc) [inline .dumi/theme/slots/Sidebar/index.tsx:153]:
```suggestion
    <MobileMenu key="Mobile-menu">{menuChild}</MobileMenu>
```

File: .dumi/theme/slots/Sidebar/index.tsx (hunk 1 above)
Comment (MadCcc) [inline .dumi/theme/slots/Sidebar/index.tsx:156]:
```suggestion
      <section className="main-menu-inner">{menuChild}</section>
```

File: components/layout/__tests__/token.test.tsx [hunk 4]
```diff
@@ -24,8 +24,7 @@ describe('Layout.Token', () => {
       >
         <Header>
           <Menu
-            theme="dark"
-            mode="horizontal"
+            mode='horizontal'
```
Comment (MadCcc) [inline components/layout/__tests__/token.test.tsx:27]:
```suggestion
            mode="horizontal"
```

File: components/layout/__tests__/token.test.tsx [hunk 5]
```diff
@@ -62,8 +61,7 @@ describe('Layout.Token', () => {
...
```
Comment (MadCcc) [inline components/layout/__tests__/token.test.tsx:64]:
```suggestion
            mode="horizontal"
```

File: components/menu/demo/component-token.tsx [hunk 6]
```diff
@@ -52,14 +61,50 @@ const items: MenuProps['items'] = [
   },
   {
     label: (
-      <a href="https://ant.design" target="_blank" rel="noopener noreferrer">
+      <a href='https://ant.design' target='_blank' rel='noopener noreferrer'>
```
Comment (MadCcc) [inline components/menu/demo/component-token.tsx:64]:
```suggestion
      <a href="https://ant.design" target="_blank" rel="noopener noreferrer">
```

File: components/menu/demo/component-token.tsx [hunk 7]
```diff
@@ -69,18 +114,44 @@ const App: React.FC = () => {
   };
 
   return (
-    <ConfigProvider
-      theme={{
-        components: {
-          Menu: {
-            horizontalItemBorderRadius: 6,
-            horizontalItemHoverBg: '#f5f5f5',
+    <Space direction='vertical'>
```
Comment (MadCcc) [inline components/menu/demo/component-token.tsx:117]:
```suggestion
    <Space direction="vertical">
```

File: components/menu/demo/component-token.tsx [hunk 8]
```diff
@@ -69,18 +114,44 @@ const App: React.FC = () => {
...
+        theme={{
+          components: {
+            Menu: {
+              horizontalItemBorderRadius: 6,
+              horizontalItemHoverBg: '#f5f5f5',
+            },
+          },
+        }}
+      >
+        <Menu onClick={onClick} selectedKeys={[current]} mode='horizontal' items={items} />
```
Comment (MadCcc) [inline components/menu/demo/component-token.tsx:128]:
```suggestion
        <Menu onClick={onClick} selectedKeys={[current]} mode="horizontal" items={items} />
```

**Comment by MadCcc** [comment]
> **The latest updates on your projects.** Learn more about [Argos notifications ↗︎](https://argos-ci.com/docs/notifications)
> 
> Build	Status	Details	Updated (UTC)
> **default** ([Inspect](https://app.argos-ci.com/ant-design/ant-design/builds/9995))	🧿 Changes detected ([Review](https://app.argos-ci.com/ant-design/ant-design/builds/9995))	1 change	Jul 17, 2023, 1:59 AM

Expected

**Pull request merged**